### PR TITLE
Team statistics module feature complete

### PR DESCRIPTION
**Pull request by c-bartell**
- Created TeamStats module
    - `team_info`, `best_season`, `worst_season`, `average_win_percentage`, `most_goals_scored`, `fewest_goals_scored`, `favorite_opponent`, `rival`
- Created associated tests
    - All tests passing, 100% coverage
- Passes spec harness tests
- Tests still need to be refactored with mocks and stubs
- Many methods still need helpers extracted

### CODE REVIEW

File: lib/team.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Team
   attr_reader :team_id, :franchise_id, :team_name, :abbreviation, :stadium, :link
   def initialize(row)
-    @team_id = row["team_id"].to_i
-    @franchise_id = row["franchiseId"].to_i
+    @team_id = row["team_id"]
+    @franchise_id = row["franchiseId"]
```
Comment (s-steel) [inline lib/team.rb]: This data type will be different for one of the branches.  We can change that later.

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        season_hash[game.season] = [game]
+      else
+        season_hash[game.season] << game
+      end
+    end
+
+    season_hash
+  end
+
+  def result_counts_by_team_id(team_id)  # Refactor to take game_ids as an arg
```
Comment (s-steel) [inline lib/team_statistics.rb:41]: I like the notes.  Should definitely look at refactoring.

File: lib/team_statistics.rb [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      opponent_games[team_id][:win_percent] = (opponent_games[team_id][:wins]/opponent_games[team_id][:total].to_f).round(2)
+    end
+
+    rival_id = opponent_games.keys.max_by do |team_id|
+      opponent_games[team_id][:win_percent]
+    end
+
+    team_info(rival_id)["team_name"]
+  end
+end
```
Comment (s-steel) [inline lib/team_statistics.rb:186]: Will need to refactor these last few methods.

File: test/team_statistics_test.rb [hunk 4]
```diff
@@ -0,0 +1,144 @@
+require './test/test_helper'
+require './lib/stat_tracker'
+
+class TeamStatisticsTest < Minitest::Test
```
Comment (s-steel) [inline test/team_statistics_test.rb:4]: I like the amount of tests and agree with the note about maybe adding more assertions.

**Review by s-steel — APPROVED** [review]
Looks good, just will need some refactoring.

File: lib/team_statistics.rb [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    games_array.each do |game|
+      if season_hash[game.season] == nil
+        season_hash[game.season] = [game]
+      else
+        season_hash[game.season] << game
+      end
+    end
+
+    season_hash
+  end
```
Comment (GregJMitchell) [inline lib/team_statistics.rb:39]: I like how many helper methods you make

**Review by GregJMitchell — APPROVED** [review]
Overall great work! We can for sure refactor, but this is great work

**Pull request merged**